contracts/0.7.x/MultipleArbitrableTokenTransactionWithAppeals.sol

```diff
@@ -439,24 +436,25 @@ contract MultipleArbitrableTokenTransactionWithAppeals is IArbitrable, IEvidence
     function fundAppeal(uint256 _transactionID, Transaction calldata _transaction, Party _side) public payable onlyValidTransactionCD(_transactionID, _transaction) {
         require(_side == Party.Sender || _side == Party.Receiver, "Wrong party.");
         require(_transaction.status == Status.DisputeCreated, "No dispute to appeal");
-        require(arbitrator.disputeStatus(_transaction.disputeID) == IArbitrator.DisputeStatus.Appealable, "Dispute is not appealable.");
 
         (uint256 appealPeriodStart, uint256 appealPeriodEnd) = arbitrator.appealPeriod(_transaction.disputeID);
         require(block.timestamp >= appealPeriodStart && block.timestamp < appealPeriodEnd, "Funding must be made within the appeal period.");
 
-        uint256 winner = arbitrator.currentRuling(_transaction.disputeID);
         uint256 multiplier;
-        if (winner == uint256(_side)){
-            multiplier = winnerStakeMultiplier;
-        } else if (winner == 0){
-            multiplier = sharedStakeMultiplier;
-        } else {
-            require(block.timestamp - appealPeriodStart < (appealPeriodEnd - appealPeriodStart)/2, "The loser must pay during the first half of the appeal period.");
-            multiplier = loserStakeMultiplier;
+        {
+            uint256 winner = arbitrator.currentRuling(_transaction.disputeID);
+            if (winner == uint256(_side)){
+                multiplier = winnerStakeMultiplier;
+            } else if (winner == 0){
+                multiplier = sharedStakeMultiplier;
+            } else {
+                require(block.timestamp - appealPeriodStart < (appealPeriodEnd - appealPeriodStart)/2, "The loser must pay during the first half of the appeal period.");
+                multiplier = loserStakeMultiplier;
+            }
         }
 
         Round storage round = roundsByTransactionID[_transactionID][roundsByTransactionID[_transactionID].length - 1];
-        require(!round.hasPaid[uint256(_side)], "Appeal fee has already been paid.");
+        require(_side != round.sideFunded, "Appeal fee has already been paid.");
```

Comment:
The previous line. This always checks the sideFunded of the current (last) round. When the appeal is funded a new empty round is created.

Reply:
In other words, you can safely clear sideFunded after the round has been appealed, because you don't need that information anymore

Reply:
But then you can fund appeals in advance? I guess that's fine?

Reply:
What do you mean in advance?

Reply:
Before the current round is finished.

Reply:
I can’t see anything wrong here, but maybe I’m missing sth. You can’t fund appeals in advance, because the new round is pushed only after the appeal has been created and you are allowed to fund only during the appeal period.

Reply:
Yeah, I think the appeal period protects this. Not sure if we want to rely on the arbitrator behaving correctly there, but I guess it is trusted.